fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java

```diff
@@ -871,6 +871,9 @@ public void releaseBackends(String clusterName, boolean isReplay) {
         List<Backend> backends = Lists.newArrayList();
         for (List<Backend> list : backendMaps.values()) {
             if (FeConstants.runningUnitTest || Config.allow_replica_on_same_host) {
+                if (needAlive) {
```

Comment:
This block:
```
if (needAlive) {
    list = list.stream().filter(Backend::isAlive).collect(Collectors.toList());
}
```

should be added to the following `else` clause:

```
} else {
    if (needAlive) {
        list = list.stream().filter(Backend::isAlive).collect(Collectors.toList());
    }
    Collections.shuffle(list);
    backends.add(list.get(0));
}
```

Reply:
Sorry for not having noticed the code have already changed too much, i have updated and tested.